tgbot/src/api/auth.py

```diff
@@ -0,0 +1,98 @@
+import typing as tp
+
+from collections import namedtuple
+
+from aiohttp import ClientResponse
+
+from api.client import client
+
+# ------------------------------------------------------------------------------
+
+Session = namedtuple("Session", ["id", "code", "chat_id"])
+
+
+async def fetch_session(params: tp.Dict[str, tp.Any]) -> tp.Optional[Session]:
+    """Fetch session for chat. Session may not exist, hence `Optional`."""
+
+    async with client.get("session", params=params) as response:
+        data: tp.List[tp.Dict[str, tp.Any]] = await response.json()
+
```

Comment:
In Pyhton 3.9, we can use `params: dict[...]` instead of `params: tp.Dict[...]` (PEP: 585)

Reply:
The same applies for lists btw

Reply:
Fixed.